### PR TITLE
Add Access Module

### PR DESCRIPTION
**Pull request by Mysterious-Dev**
This PR add a new module with somes AW to let modder access to register functions

- [x] Villager Profession
- [x] SpawnRestriction

WIP

### CODE REVIEW

File: fabric-access-v1/src/main/resources/fabric-access-v1.accesswidener [hunk 1]
```diff
@@ -0,0 +1,11 @@
+accessWidener	v1	named
```
Comment (deirn) [inline fabric-access-v1/src/main/resources/fabric-access-v1.accesswidener:1]:
```suggestion
accessWidener	v2	named
```

You need v2 for transitive access widener support.

File: fabric-access-v1/src/main/resources/fabric-access-v1.accesswidener [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+accessible    method    net/minecraft/entity/SpawnRestriction    register    (Lnet/minecraft/entity/EntityType;Lnet/minecraft/entity/SpawnRestriction$Location;Lnet/minecraft/world/Heightmap$Type;Lnet/minecraft/entity/SpawnRestriction$SpawnPredicate;)V
```
Comment (deirn) [inline fabric-access-v1/src/main/resources/fabric-access-v1.accesswidener:3]:
```suggestion
transitive-accessible    method    net/minecraft/entity/SpawnRestriction    register    (Lnet/minecraft/entity/EntityType;Lnet/minecraft/entity/SpawnRestriction$Location;Lnet/minecraft/world/Heightmap$Type;Lnet/minecraft/entity/SpawnRestriction$SpawnPredicate;)V
```
Need the prefix so projects that depend on this module can actually access them :P.

**Review by deirn — CHANGES_REQUESTED** [review]

File: fabric-access-v1/src/main/resources/fabric-access-v1.accesswidener (hunk 2 above)
Comment (Mysterious-Dev) [reply fabric-access-v1/src/main/resources/fabric-access-v1.accesswidener:3]: It's needed for all actual AW ?

Comment (deirn) [reply fabric-access-v1/src/main/resources/fabric-access-v1.accesswidener:3]: I guess, yeah? You don't need to prefix it if it is just for internal use.
I suggested one change just for example.

**Comment by Juuxel** [comment]
This was already implemented in an actually functional way (using AW v2 and transitive access wideners) in #2044. These registration methods can become a new PR extending that if needed.

**Pull request closed without merging**